http://dev.mutt.org/trac/changeset/25e46aad362b#file0

Do not free statically-allocated buffy buffer.
Unbreaks unmailboxes, closes #3273.

Index: buffy.c
===================================================================
--- buffy.c (revision 5851:8b2af5b64981)
+++ buffy.c (revision 5914:25e46aad362b)
@@ -237,5 +237,4 @@
       if(*tmp)
       {
-        FREE (&((*tmp)->path));
         tmp1=(*tmp)->next;
         FREE (tmp);		/* __FREE_CHECKED__ */
